src/simulations/PowerPerturbation.jl

```diff
@@ -3,85 +3,30 @@ using Setfield
 
 """
 ```Julia
-PowerPerturbation(;node_number,fraction,tspan_fault)
+PowerPerturbation(;node_number,power_new,tspan_fault,var)
 ```
 # Keyword Arguments
-- `node_number`: number  of the node
-- `fraction`: percentage factor to be applied to the active power P
+- `node_number`: number or name of the node
+- `power_new`: value of power during fault event
 - `tspan_fault`: PowerPerturbation timespan
-- `power_symbol`: parameter symbol on the node that represents Power, default is :P
+- `var`: parameter symbol on the node that represents Power, default is :P
 """
-Base.@kwdef struct PowerPerturbation
-    node_number
-    fraction
-    tspan_fault
-    power_symbol = :P
-end
 
-"Error to be thrown if something goes wrong during power perturbation"
-struct PowerPerturbationError <: PowerDynamicsError
-    msg::String
+Base.@kwdef struct PowerPerturbation <:NodePerturbation
+    node
+    power_new
+    tspan_fault
+    var = :P
 end
 
 function (pd::PowerPerturbation)(powergrid)
-    mapPowerField(powergrid, pd, p -> p * pd.fraction)
-end
-
-function mapPowerField(powergrid, pd, f)
-    node_list_power_drop = copy(powergrid.nodes)
-    node_for_drop = node_list_power_drop[pd.node_number]
-    if !(hasproperty(node_for_drop,pd.power_symbol))
-        throw(PowerPerturbationError("Node number: $(pd.node_number) must have a power parameter $(pd.power_symbol)"))
-    end
-    lens = Setfield.PropertyLens{pd.power_symbol}()
-    node_for_drop = Setfield.modify(f, node_for_drop, lens)
-    node_list_power_drop[pd.node_number] = node_for_drop
-    PowerGrid(node_list_power_drop, powergrid.lines)
+    mapField(powergrid, pd, p -> p*0.0+convert(Float64, pd.power_new))
```

Comment:
@luap-pik : the tests for the power perturbation script are failing. I used a simple trick here to overcome to problem of having integers and floats for a parameter. It worked in my test example but the specific test with two SwingEqLVS nodes is running through some numeric issue.

Reply:
I'm not sure if I understand the problem here. why do you convert to Float64, when someone passes an Int, this should be converted automatically when the rhs is called?
And in Setfield.jl you don't need to have the same type after the change.

Reply:
This conversion did not work automatically if you do not set a fraction of parameter change (previous implementation: multiplying initial value of parameter with fraction) but reset the absolute value of the parameter (the new way I implemented it)

Reply:
This sounds very weird. Why does this not happen for the other perturbations?
Is the Problem that the type of the rhs changes in the fault callback?

I noticed you write `p -> p*0.0+convert(Float64, pd.power_new)`. 
shouldn't it be enough to do `p -> convert(Float64, pd.power_new)`?

Reply:
Actually, we could do sth like `convert(eltype(getfield(np_node, np.var)), new_value)` in `mapField` to enforce type stability. This is better then hardcoding `Float64`.

Reply:
you are welcome to try our your suggestion and whether this fixes the tests ;)